website_anchor_smooth_scroll/static/src/js/website_anchor_smooth_scroll.js

```diff
@@ -1,25 +1,29 @@
 /* © 2016 Antiun Ingeniería S.L. - Jairo Llopis
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+odoo.define('website_anchor_smooth_scroll.website_anchor_smooth_scroll', function (require) {
 
+require('web.dom_ready');
 "use strict";
 
 function website_anchor_smooth_scroll(event) {
     event.preventDefault();
 
-    var target = $(event.target.hash);
-
-    return $('html, body')
-    .stop()
-    .animate({
-        'scrollTop': target.offset().top - 100
-    })
-    .promise()
-    .done(function(element) {
-        history.pushState(null, document.title, event.target.hash);
-    });
+    if (event.target.hash != "") {
+      var target = $(event.target.hash);
+      return $('html, body')
+        .stop()
+        .animate({
+            'scrollTop': target.offset().top - 100
+        })
+        .promise()
+        .done(function(element) {
+            history.pushState(null, document.title, event.target.hash);
+        });
+    };
 }
 
-(function ($) {
+$(document).ready(function () {
```

Comment:
Just put the `require('web.dom_ready');` line above this and remove this callback.